core/classifiers/BaseClassifier/BaseClassifier.py

```diff
@@ -0,0 +1,87 @@
+# coding: utf-8
+#
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Base class for classification algorithms"""
+
+import abc
+
+
+class BaseClassifier(object):
+    """A base class for classifiers that uses supervised learning to match
+    free-form text answers to answer groups. The classifier trains on answers
+    that exploration editors have assigned to an answer group. Given a new
+    answer, it predicts the answer group.
+
+    Below are some concepts used in this class.
+    training_data: list(dict). The training data that is used for training
+        the classifier. This field is populated lazily when the job request
```

Comment:
Yup. Actually, I didn't verify the doc strings in much detail before issuing PR. I copied the relevant code and took a brief look at the doc string for consistency. I guess I'll have to go through them once again.